api/src/types.rs

```diff
@@ -275,6 +290,147 @@ pub struct Backup {
     pub region: String,
     pub status: String,
     pub created: String,
+    pub linked_resource: Option<String>,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct JobDataScheduledBackup {
+    pub description: String,
+    pub max_backup_count: i32,
+    pub cluster_id: ClusterId,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct OtherJobData {
```

Comment:
It think it would be simpler (code wise) to have `JobData` defined that way:

```rust
pub struct JobData {
    pub r#type: String,
    pub data: serde_json::Value,
}
``` 
and define that instead:

```rust

impl JobData {
    pub fn decode<A: Deserialize>(self) -> serde_json::Result<A> {
        // to parsing based on the type and the data field using serde_json::from_value
    }
}
```

Reply:
Updated this to stop handling the type being wrong, since we're in control of the release cycle, will ultimately use ESC to test the platform and can avoid that.